### DISCUSSION TITLE
Question: Is there a setting to remove a vulnerability database from being used?

### DISCUSSION THREAD

**Discussion by BrittanyBoles7** [aquasecurity/trivy#5355] [Q&A] [answered]
### Question

I am trying to be able to set Trivy to only use NVD data, or Github advisory data etc.)  Is there anyway to do this? I have tried to build a Trivy-db with only NVD data but only succeeded in getting the entire vulnerability data source Trivy uses downloaded into an air gap environment. I know there is settings to filter out specific vulnerabilities, is there a setting for entire vulnerability databases? 

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

None

### Operating System

Linux

### Version

```bash
V0.45.0
```

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @BrittanyBoles7 

We don't have setting for `trivy-db` filtering.

> I have tried to build a Trivy-db with only NVD data but only succeeded in getting the entire vulnerability data source

We don't use NVD to get advisories. It is only needed  for data sources (e.g. `descriptions`, `links`, etc.). 
You can try to build GHSA + NVD. It should work.

Regards, Dmitriy

**Reply by BrittanyBoles7** [reply]
Thank you for the information! I am doing research on static analysis tools for grad school. As a follow up question, is having the NVD database in the local trivy-db a requirement? If I just wanted GHSA's and I build a trivy-db with just those would there be any problems not having NVD data?

**Reply by DmitriyLewen** [reply]
We take the following fields for NVD - https://github.com/aquasecurity/trivy-db/blob/4fc651f7ac8df621b904367e154800a1d335f6f7/pkg/vulnsrc/nvd/nvd.go#L101
We can overwrite or add data from these fields to advisories.
Then if you build trivy-db with GHSA only - advisory with NVD and advisory without NVD can be different.

Take a look on this [function](https://github.com/aquasecurity/trivy-db/blob/4fc651f7ac8df621b904367e154800a1d335f6f7/pkg/vulndb/db.go#L126) to better understand where we are using NVD.

**Reply by BrittanyBoles7** [reply]
Awesome! Thank you, this improved my understanding.
